trl/models/modeling_base.py

```diff
@@ -677,3 +677,55 @@ def create_reference_model(
         logging.warning("Pattern passed or found, but no layers matched in the model. Check for a typo.")
 
     return ref_model.eval()
+
+
+class GeometricMixtureWrapper(GenerationMixin):
+    r"""
+    Geometric Mixture module that samples from the logits of two model's geometric mixture.
+
+    Args:
+        model (`PreTrainedModel`):
+        ref_model (`PreTrainedModel`):
+        beta (`float`, *optional* - default: 0.5):
+    """
+
+    main_input_name = "input_ids"
+    _supports_cache_class = False
+    _supports_static_cache = False
+
+    def __init__(self, model, ref_model, beta=0.5, device=None):
+        super().__init__()
+
+        self.model = model.eval()
+        self.config = model.config
+        self.ref_model = ref_model.eval()
+        self.beta = beta
+        self.device = device
+
+    def __call__(self, *args, **kwargs):
+        return self.forward(*args, **kwargs)
+
+    @torch.no_grad()
+    def forward(self, *args, **kwargs):
+        model_outputs = self.model(*args, **kwargs)
+        model_logits = model_outputs.logits
+        ref_model_logits = self.ref_model(*args, **kwargs).logits
+
+        model_outputs.logits = torch.nn.functional.log_softmax(
+            self.beta * model_logits + (1 - self.beta) * ref_model_logits, dim=-1
```

Comment:
As of the image below:

![image](https://github.com/user-attachments/assets/08060952-3ebc-48fb-94f7-8acf28c21009)


```suggestion
            self.beta * ref_model_logits + (1 - self.beta) * model_logits, dim=-1
```